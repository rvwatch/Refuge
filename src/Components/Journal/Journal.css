.journal-form {
  width: 350px;
  margin: auto;
  text-align: center;
}

.journal-form input,
.journal-form textarea {
  display: block;
  margin-bottom: 15px;
  width: 100%;
}