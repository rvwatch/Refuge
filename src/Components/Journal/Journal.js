import React, {Component} from 'react';
import './Journal.css';

export class Journal extends Component {
  constructor(){
    super();
    this.state = {
      title: '',
      description: ''
    };
  }

  handleInputs = (event) => {
    const {name, value} = event.target;
    this.setState({
      [name]: value
    })
  }

  addNote = () => {
    console.log();
    
  }

  render () {
    return (
      <section>
        <header>
          <h1 className='coming-soon'>Coming Soon!</h1>
          <form className='journal-form'>
            <input type='text' placeholder='Title' value={this.state.title} />
            <textarea placeholder='Description' value={this.state.description}></textarea>
            <button>add entry</button>
          </form>
        </header>
        <section>
          <article className='note-wrap'>
            <h1>Title</h1>
            <p>Description text</p>
            <button className='delete-note'>x</button>
          </article>
        </section>
      </section>
    ); 
  }
}

