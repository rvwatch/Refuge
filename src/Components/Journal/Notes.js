import React from 'react';
import './Journal.css';

const Notes = ({ desc }) => {
  return (
    <article className='note-wrap'>
      <p>{desc}</p>
    </article>
  )
}

export default Notes;