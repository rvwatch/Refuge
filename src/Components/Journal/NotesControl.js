import React, { Component } from 'react';
import './Journal.css';

class NotesControl extends Component {
  constructor(props) {
    super(props);
    this.state = {
      desc: ''
    };
  }

  handleInput = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.addNote(this.state);
  }

  render() {
    return (
      <form className='journal-control-form' style={{textAlign:'left'}} onSubmit={this.handleSubmit} >
        <label>
          <textarea
            onChange={ this.handleInput }
            name="desc"
            type="text"
            value={ this.state.desc }
            placeholder="journal entry">
          </textarea>
        </label>
        <button type="submit">Submit</button>
      </form>
    );
  }
}

export default NotesControl;
