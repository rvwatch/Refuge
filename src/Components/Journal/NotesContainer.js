import React from 'react';
import './Journal.css';
import Notes from './Notes';

const NotesContainer = ({notes}) => {
  const renderNotes = notes.length ? notes.map(note => <Notes {...note} key={note.title} />) : '';
  return (
    <section>
      {renderNotes}
    </section>
  )
};

export default NotesContainer;