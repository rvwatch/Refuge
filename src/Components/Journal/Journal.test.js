import  React from 'react';
import {shallow} from 'enzyme';
import {Journal} from './Journal';

describe('Journal', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Journal />);
    expect(wrapper).toMatchSnapshot();
  });
});
