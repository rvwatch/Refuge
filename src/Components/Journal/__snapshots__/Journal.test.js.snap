// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Journal should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Journal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1
                    className="coming-soon"
          >
                    Coming Soon!
          </h1>
          <form
                    className="journal-form"
          >
                    <input
                              placeholder="Title"
                              type="text"
                              value=""
                    />
                    <textarea
                              placeholder="Description"
                              value=""
                    />
                    <button>
                              add entry
                    </button>
          </form>
</header>,
        <section>
          <article
                    className="note-wrap"
          >
                    <h1>
                              Title
                    </h1>
                    <p>
                              Description text
                    </p>
                    <button
                              className="delete-note"
                    >
                              x
                    </button>
          </article>
</section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="coming-soon"
>
              Coming Soon!
</h1>,
            <form
              className="journal-form"
>
              <input
                            placeholder="Title"
                            type="text"
                            value=""
              />
              <textarea
                            placeholder="Description"
                            value=""
              />
              <button>
                            add entry
              </button>
</form>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Coming Soon!",
              "className": "coming-soon",
            },
            "ref": null,
            "rendered": "Coming Soon!",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  placeholder="Title"
                  type="text"
                  value=""
/>,
                <textarea
                  placeholder="Description"
                  value=""
/>,
                <button>
                  add entry
</button>,
              ],
              "className": "journal-form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "placeholder": "Title",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "placeholder": "Description",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "add entry",
                },
                "ref": null,
                "rendered": "add entry",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <article
            className="note-wrap"
>
            <h1>
                        Title
            </h1>
            <p>
                        Description text
            </p>
            <button
                        className="delete-note"
            >
                        x
            </button>
</article>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Title
</h1>,
              <p>
                Description text
</p>,
              <button
                className="delete-note"
>
                x
</button>,
            ],
            "className": "note-wrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Title",
              },
              "ref": null,
              "rendered": "Title",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Description text",
              },
              "ref": null,
              "rendered": "Description text",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "x",
                "className": "delete-note",
              },
              "ref": null,
              "rendered": "x",
              "type": "button",
            },
          ],
          "type": "article",
        },
        "type": "section",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1
                        className="coming-soon"
            >
                        Coming Soon!
            </h1>
            <form
                        className="journal-form"
            >
                        <input
                                    placeholder="Title"
                                    type="text"
                                    value=""
                        />
                        <textarea
                                    placeholder="Description"
                                    value=""
                        />
                        <button>
                                    add entry
                        </button>
            </form>
</header>,
          <section>
            <article
                        className="note-wrap"
            >
                        <h1>
                                    Title
                        </h1>
                        <p>
                                    Description text
                        </p>
                        <button
                                    className="delete-note"
                        >
                                    x
                        </button>
            </article>
</section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="coming-soon"
>
                Coming Soon!
</h1>,
              <form
                className="journal-form"
>
                <input
                                placeholder="Title"
                                type="text"
                                value=""
                />
                <textarea
                                placeholder="Description"
                                value=""
                />
                <button>
                                add entry
                </button>
</form>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Coming Soon!",
                "className": "coming-soon",
              },
              "ref": null,
              "rendered": "Coming Soon!",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    placeholder="Title"
                    type="text"
                    value=""
/>,
                  <textarea
                    placeholder="Description"
                    value=""
/>,
                  <button>
                    add entry
</button>,
                ],
                "className": "journal-form",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "placeholder": "Title",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "placeholder": "Description",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "add entry",
                  },
                  "ref": null,
                  "rendered": "add entry",
                  "type": "button",
                },
              ],
              "type": "form",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <article
              className="note-wrap"
>
              <h1>
                            Title
              </h1>
              <p>
                            Description text
              </p>
              <button
                            className="delete-note"
              >
                            x
              </button>
</article>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Title
</h1>,
                <p>
                  Description text
</p>,
                <button
                  className="delete-note"
>
                  x
</button>,
              ],
              "className": "note-wrap",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Title",
                },
                "ref": null,
                "rendered": "Title",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Description text",
                },
                "ref": null,
                "rendered": "Description text",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "x",
                  "className": "delete-note",
                },
                "ref": null,
                "rendered": "x",
                "type": "button",
              },
            ],
            "type": "article",
          },
          "type": "section",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
